### DISCUSSION TITLE
could not connect to server: Connection refused

### DISCUSSION THREAD

**Discussion by Aref20** [saleor/saleor#10281] [Deployment]
I faced this error after running docker run -p 8000:8000 --env-file .env --name saleor-core saleor on saleor api

**could not connect to server: Connection refused
	Is the server running on host "localhost" (127.0.0.1) and accepting
	TCP/IP connections on port 5432?
could not connect to server: Cannot assign requested address
	Is the server running on host "localhost" (::1) and accepting
	TCP/IP connections on port 5432?**

Can anyone help me?

**Comment by karolkielecki** [comment]
Hi @Aref20, have you tried the solutions provided here? https://stackoverflow.com/questions/37307346/is-the-server-running-on-host-localhost-1-and-accepting-tcp-ip-connections

**Reply by Aref20** [reply]
should I create the docker-compose file and add the PostgreSQL database to it?

**Reply by Aref20** [reply]
or just run the docker file?

**Reply by patrys** [reply]
You will need a running instance of PostgreSQL for Saleor to communicate with. It does not have to be dockerized, but you need to ensure Docker can reach it over the network and that `DATABASE_URL` is configured correctly.
